File: 1470. Shuffle the Array/main.c
#include <stdio.h>

/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int *shuffle(int *nums, int numsSize, int n, int *returnSize) {
    int *res = (int *)calloc(2 * n, sizeof(int));
    if (res == NULL) {
        return NULL;
    }

    for (int i = 0; i < n; i++) {
        res[2 * i] = nums[i];
        res[2 * i + 1] = nums[n + i];
    }

    *returnSize = 2 * n;
    return res;
}

File: 1470. Shuffle the Array/main.rs
struct Solution;

impl Solution {
    pub fn shuffle(nums: Vec<i32>, n: i32) -> Vec<i32> {
        let mut res = Vec::new();

        for i in 0..n as usize {
            res.push(nums[i]);
            res.push(nums[i + n as usize]);
        }

        res
    }
}

impl Solution {
    pub fn shuffle(nums: Vec<i32>, n: i32) -> Vec<i32> {
        (0..2 * n as usize)
            .map(|i| {
                if i % 2 == 0 {
                    nums[i / 2]
                } else {
                    nums[i / 2 + n as usize]
                }
            })
            .collect()
    }
}